win/rl/winrl.cc

```diff
@@ -177,10 +177,14 @@ class NetHackRL
     std::array<uint8_t, (COLNO - 1) * ROWNO> chars_;
     std::array<uint8_t, (COLNO - 1) * ROWNO> colors_;
     std::array<uint8_t, (COLNO - 1) * ROWNO> specials_;
-
+    
+    typedef std::array<char, NLE_GLYPH_STR_LENGTH + 1> glyph_str_t;
```

Comment:
I have a slight preference for more C-like code here but I guess it doesn't matter too much. However, I'm not certain `NLE_GLYPH_STR_LENGTH + 1` is enough. All call sites of `do_screen_description` give it a `char out_str[BUFSZ]`, and BUFSZ is going to be at least 1024. I suspect bad things to happen if `out_str` has too little space, and there's no way for `do_screen_description` to tell. I think the truncation should be done when we copy it for the observation itself.

Reply:
truncation happening at copying for observation.